topydo/lib/PrettyPrinterFilter.py

```diff
@@ -57,15 +60,21 @@ def filter(self, p_todo_str, p_todo):
             except KeyError:
                 pass
 
-            p_todo_str = color + p_todo_str + NEUTRAL_COLOR
-
+            p_todo_str = '%s%s%s' % (color, p_todo_str, NEUTRAL_COLOR)
             if config().highlight_projects_contexts():
                 p_todo_str = re.sub(
                     r'\B(\+|@)(\S*\w)',
-                    PROJECT_COLOR + r'\g<0>' + color,
+                    #PROJECT_COLOR + r'\g<0>' + color,
+                    lambda m: (
+                        CONTEXT_COLOR if m.group(0)[0] == "+"
+                        else PROJECT_COLOR)+m.group(0)+color,
                     p_todo_str)
-
-            p_todo_str += NEUTRAL_COLOR
+            p_todo_str = re.sub(r'\w(\S*:[^/\s]\S*)\w',METADATA_COLOR+r'\g<0>'+color,p_todo_str)
```

Comment:
You;re right, but it's essentially the same here, due to the `\g<0>` backreference.
From [7.2. re — Regular expression operations — Python 2.7.10rc0 documentation](https://docs.python.org/2/library/re.html) : 

    The backreference \g<0> substitutes in the entire substring matched by the RE.